### PR TITLE
Including documentation for radio element in interactive dialog

### PR DESCRIPTION
**Pull request by kaakaa**
#### Summary
Add description about `radio` type in interactive dialog.

#### Ticket Link
https://github.com/mattermost/docs/issues/3589

### CODE REVIEW

**Comment by kaakaa** [comment]
As for [the explanation of `options`](https://github.com/mattermost/docs/pull/3635/files#diff-2cd2701b686698512b36aebea072fd15R282), I am using [the explanation of the existing Select Element](https://github.com/mattermost/docs/pull/3635/files#diff-2cd2701b686698512b36aebea072fd15R209), but isn't the type of `options` an `Array` instead of a `String`?

>     "options", "String", "(Optional) An array of options for the radio element."

If it looks good to you, I can fix it, but what do you think?

File: source/developer/interactive-dialogs.rst [hunk 1]
```diff
@@ -240,6 +242,48 @@ The full list of supported fields are included below:
...
+        "value": "engineering"
+      },
+      {
+        "text": "Sales",
+        "value": "sales"
+      },
+      {
+        "text": "Administration",
+        "value": "administration"
+      }
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst:270]: Given that the image above as a `default` element, it should also be added to this payload.

File: source/developer/interactive-dialogs.rst [hunk 2]
```diff
@@ -240,6 +242,48 @@ The full list of supported fields are included below:
...
+    ]
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst:271]: Non blocking: I would also love to see the example above being extended with a `help_text` to ensure that every supported field is shown.

File: source/developer/interactive-dialogs.rst [hunk 3]
```diff
@@ -66,6 +66,8 @@ Each dialog supports elements for users to enter information.
 - **text**: Single-line plain text field. Use this for inputs such as names, email addresses or phone numbers.
 - **textarea**: Multi-line plain text field. Use this field when the answer is expected to be longer than 150 characters. 
 - **select**: Message menu. Use this for pre-selected choices. Can either be static menus or dynamic menus generated from users and public channels of the system. For more information on message menus, see :doc:`the documentation <interactive-messages>`.
+- **bool**: Checkbox option. Use this for binary selection.
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst:69]: Are you also interested in documenting this type?

File: source/developer/interactive-dialogs.rst [hunk 4]
```diff
@@ -240,6 +242,48 @@ The full list of supported fields are included below:
...
+
+The full list of supported fields are included below:
+
+.. csv-table::
+    :header: "Field", "Type", "Description"
+
+    "display_name", "String", "Display name of the field shown to the user in the dialog. Maximum 24 characters."
+    "name", "String", "Name of the field element used by the integration. Maximum 300 characters. You should use unique “name” fields in the same dialog."
+    "type", "String", "Set this value to ``radio`` for a radio element."
+    "options", "String", "(Optional) An array of options for the radio element."
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst:282]: Correct, that is supposed to be an `Array`

File: source/developer/interactive-dialogs.rst (hunk 3 above)
Comment (kaakaa) [reply source/developer/interactive-dialogs.rst:69]: No, documentation about `bool` type had already been written.
https://docs.mattermost.com/developer/interactive-dialogs.html#checkbox-element

When writing about `radio type`, I noticed that `bool` is not here, so I just added it.

**Comment by kaakaa** [comment]
Thanks @hanzei! I've updated as your comment 👍

File: source/developer/interactive-dialogs.rst (hunk 3 above)
Comment (hanzei) [reply source/developer/interactive-dialogs.rst:69]: Missed that, thanks for the explanation :+1:

File: source/developer/interactive-dialogs.rst [hunk 5]
```diff
@@ -213,7 +215,7 @@ The list of supported fields for the ``select`` type element is included below:
 Checkbox Element
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-In Mattermost version 5.16 and later, you can use checkbox elements. It looks like a plain text field with a checkbox to be selected. Below is an example of a checkbox element that asks for meeting feedback:
+In Mattermost version 5.16 and later, you can use checkbox elements. It looks like a plain text field with a checkbox to be selected. Below is an example of a checkbox element that asks for meeting feedback.
```
Comment (justinegeffen) [inline source/developer/interactive-dialogs.rst:218]:
```suggestion
From Mattermost 5.16 you can use checkbox elements. It looks like a plain text field with a checkbox to be selected. Below is an example of a checkbox element that asks for meeting feedback.
```

File: source/developer/interactive-dialogs.rst [hunk 6]
```diff
@@ -240,6 +242,50 @@ The full list of supported fields are included below:
     "placeholder", "String", "(Optional) A string displayed to include a label besides the checkbox. Maximum 150 characters."
     
 
+Radio Element
+^^^^^^^^^^^^^^^^^^^^^^^
+
+In Mattermost version 5.16 and later, you can use radio elements. It looks like a plain text field with a radio button to be selected. Below is an example of a radio element that asks for a department.
```
Comment (justinegeffen) [inline source/developer/interactive-dialogs.rst:248]:
```suggestion
From Mattermost 5.16 you can use radio elements. It looks like a plain text field with a radio button to be selected. Below is an example of a radio element that asks for a department.
```

**Comment by justinegeffen** [comment]
Thanks, @kaakaa. This is super-helpful. :)

**Pull request merged**